requestor/src/test/java/TestAgeStringBuilder.java

```diff
@@ -0,0 +1,80 @@
+import de.rwth.imi.flare.api.model.*;
+import de.rwth.imi.flare.api.model.mapping.MappingEntry;
+import de.rwth.imi.flare.requestor.SearchQueryStringBuilder;
+import org.junit.jupiter.api.Test;
+import java.net.URISyntaxException;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.List;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestAgeStringBuilder {
+    @Test
+    public void mainAgeTests() throws URISyntaxException {
+
+        assertEquals("Patient?birthdate=lt" + getUpdatedDate("1987-11-16"), ageSingeComparisonRequest(35.0, "a", Comparator.gt));
+        assertEquals("Patient?birthdate=gt" + getUpdatedDate("1990-11-16"), ageSingeComparisonRequest(32.0, "a", Comparator.lt));
+        assertEquals("Patient?birthdate=gt" + getUpdatedDate("1990-11-16"), ageSingeComparisonRequest(32.634,  "a", Comparator.lt));
+        assertEquals("Patient?birthdate=gt" + getUpdatedDate("1990-11-16"), ageSingeComparisonRequest(32.0 * 12.0, "mo", Comparator.lt));
+        assertEquals("Patient?birthdate=lt" + getUpdatedDate("2022-11-02"), ageSingeComparisonRequest(2.0,  "wk", Comparator.gt));
+
+        assertEquals("Patient?birthdate=gt" + getUpdatedDate("1989-11-17")
+                          + "&birthdate=lt" + getUpdatedDate("1990-11-16"), ageSingeComparisonRequest(32.0,  "a", Comparator.eq));
+
+        assertEquals("Patient?birthdate=lt" + getUpdatedDate("2021-11-16") + "&birthdate=gt" + getUpdatedDate("2019-11-16")
+                , ageRangeRequest(1.0, 3.0, "a" ));
+
+        ageSingeComparisonRequest(32.0,  "a", Comparator.ne); // should produce a "comparator not implemented" exception
+        ageSingeComparisonRequest(2.0, "d", Comparator.gt); // should produce a unit exception
```

Comment:
@bastianschaffer : surround with try-catch and then assert exception thrown.
also note - that you either have to create your own exception class or check against a more generic exception in this case you could use: IllegalArgumentException